js-code/sudokuValidation.js

```diff
@@ -0,0 +1,33 @@
+arr = [
+  [0, 0, 0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 5, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0, 0, 0],
+  [0, 0, 0, 0, 0, 0, 0, 0, 0],
+];
+
+function validate(arr, r, c, val) {
+  var res = false;
```

Comment:
```res``` is a variable that has no meaning. Meaningful variables help other coders understand the syntax and code flow. Try renaming it to ```isValid```.

Reply:
using ```var``` inside a function scope is highly discouraged. 
https://www.w3schools.com/js/js_scope.asp